charts/consul/templates/connect-inject-deployment.yaml

```diff
@@ -128,12 +128,13 @@ spec:
                 -log-json={{ .Values.global.logJSON }} \
                 -default-inject={{ .Values.connectInject.default }} \
                 -consul-image="{{ default .Values.global.image .Values.connectInject.imageConsul }}" \
-                -envoy-image="{{ .Values.global.imageEnvoy }}" \
+                -consul-dataplane-image="{{ .Values.global.imageConsulDataplane }}" \
                 -consul-k8s-image="{{ default .Values.global.imageK8S .Values.connectInject.image }}" \
                 -release-name="{{ .Release.Name }}" \
                 -release-namespace="{{ .Release.Namespace }}" \
                 -resource-prefix={{ template "consul.fullname" . }} \
                 -listen=:8080 \
+                -tls-server-name="server.{{ .Values.global.datacenter }}.{{ .Values.global.domain }}" \
```

Comment:
For now we'll always need to pass TLS server name so that host verification works, but eventually consul-dataplane will infer that if the address is a DNS name